scipy/stats/_hypotests.py

```diff
@@ -9,6 +9,276 @@
 from ._continuous_distns import chi2, norm
 from scipy.special import gamma, kv, gammaln
 from . import _wilcoxon_data
+import scipy.stats.stats
+from functools import wraps
+from scipy._lib._docscrape import FunctionDoc, Parameter
+import inspect
+
+
+# TODO: add support for `axis` tuples
+def _remove_nans(samples, paired):
+    "Remove nans from paired or unpaired samples"
+    # potential optimization: don't copy arrays that don't contain nans
```

Comment:
Yes, i considered it. I think I even wrote the code for it here or in another PR, but couldn't find it. I'll consider rewriting it.

Reply:
With the current approach (using `np.apply_along_axis`), this would be a little unnatural. Since `np.apply_along_axis` is taking care of looping over the axis-slices, the easiest way to do this would be to store this boolean array in the same array (perhaps in another dimension) as the one that contains the samples themselves. If we are going to go this route, it would make more sense to go back to `np.nditer` and loop manually (which I was doing in the initial commits but moved away from). Does it make sense to you to explore this in a followup, focusing first on correctness of implementation, completeness of tests, and the overall decorator idea? It's a good potential optimization, but of course if we were most interested in speed then we'd be implementing this separately for each test.

Reply:
Yeah, when this is merged, maybe we create a meta-issue about the `axis` and `nan_policy` arguments of all `stats`/`mstats` functions to keep track of which functions do/don't have these arguments, which use this wrapper, which are natively vectorized, etc.

Ideas for improvement to this wrapper can be kept there.

Let's leave this comment unresolved as a reminder?